[platform] add `platform` folder for each board with better abstraction

enkerewpo opened the pull request:
https://wheatfox.notion.site/hvisor-platform-abstraction-1ae11a91ec7a807bb873dbefe69a7c8c

KouweiLee reviewed (commented):
![1e5cf66cd272bb91b91e03f41a744a9](https://github.com/user-attachments/assets/5be12faa-9ae2-4116-94be-bb1defe4102f)
When I run hvisor on nxp imx8mp, it had some errors and would pause for two seconds. But then hvisor and root-linux can boot successfully, I am not sure if it is a problem. @ZhongkaiXu

enkerewpo commented:
https://github.com/syswonder/hvisor/blob/dev-board-abstract/platform/aarch64/imx8mp/cargo/features
```
gicv3
imx_uart
iommu
```
I added `iommu` feature to imx8mp just like qemu-* board, but is imx8mp board not supporting `iommu`? You can remove the `iommu` feature in `platform/aarch64/imx8mp/cargo/features` then.

enkerewpo commented:
removed `iommu`: https://github.com/syswonder/hvisor/pull/121/commits/e2fb25360ee5ae95229db48c9df4606e27721acd

dallasxy commented:
aarch64 may need to add features to distinguish a55 from other CPUs so that it can distinguish the mpidr2cpuid mode when merging rk35*8 code later

enkerewpo commented:
UPDATE:
1. updated riscv64 dts
2. removed `iommu` feature from zcu102 and updated the variables in zcu102 platform.mk
3. add Makefile for all `image/dts` for dtb generation
